llama_index/llms/openai_like.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Optional
+
+from llama_index.bridge.pydantic import Field
+from llama_index.constants import DEFAULT_CONTEXT_WINDOW
+from llama_index.llms import OpenAI
+
+
+class OpenAILike(OpenAI):
+    """
+    OpenAILike is a thin wrapper around the OpenAI model that makes it compatible with 3rd party tools
+    that provide an openai-compatible api
+
+    Currently, llama_index prevents using custom models with their OpenAI class because they need to be able to infer some metadata from the model name.
+
+    NOTE: You still need to set the OPENAI_BASE_API and OPENAI_KEY environment variables. OPENAI_KEY can normally be set to anything in this case, but will depend on the tool you're using.
```

Comment:
Fwiw, this comment needs some work:
- It's `OPENAI_API_KEY` not `OPENAI_KEY`
- One can set these values using construction kwargs, it doesn't just have to be via environment variables